tests/test_xyz/test_points_poly.py

```diff
@@ -235,229 +77,107 @@ def test_points_in_polygon():
-    )
-    logger.info(fence.dataframe.to_string())
+def test_check_multi_point_single_polygon_zdir():
+    pol = Polygons(SMALL_POLY_INNER)
+    poi = Points([(4.0, 4.0, 0.0), (4.0, 4.0, 4.0), (4.0, 4.0, 8.0)])
+    assert len(poi.dataframe) == 3
 
-    assert len(fence.dataframe) == 145
-    assert fence.dataframe.H_DELTALEN.mean() == pytest.approx(0.1414, abs=0.01)
-    assert fence.dataframe.H_DELTALEN.std() <= 0.001
-    assert fence.dataframe.H_CUMLEN.max() == pytest.approx(10.0, abs=0.5)
-    assert fence.dataframe.H_CUMLEN.min() == pytest.approx(-10.0, abs=0.5)
+    # Note z-direction has no effect. All points are deleted.
+    poi.operation_polygons(pol, value=1, opname="eli", inside=True)
+    assert len(poi.dataframe) == 0
 
 
-def test_fence_from_slanted_polygon():
-    """Test fence from polygon which is slanted; but total HLEN is less than distance"""
+def test_check_multi_point_multi_polyon_inside_op():
+    pol = Polygons(SMALL_POLY_INNER + LARGE_POLY_SHIFTED)
+    # Two points in small cube, one in large cube, one outside
+    poi = Points([(4.0, 4.0, 0.0), (4.5, 4.0, 0.0), (7.0, 7.0, 0.0), (20.0, 5.0, 0.0)])
+    assert len(poi.dataframe) == 4
 
-    pol = Polygons()
+    poi.operation_polygons(pol, value=1, opname="eli", inside=True)
+    assert len(poi.dataframe) == 1
 
-    mypoly = {
-        pol.xname: [0.0, 3.0, 6.0],
-        pol.yname: [100.0, 102.0, 104.0],
-        pol.zname: [0, 50, 60],
-        pol.pname: [1, 1, 1],
-    }
-    pol.dataframe = pd.DataFrame(mypoly)
-
-    fence = pol.get_fence(
-        distance=10, nextend=1, name="SOMENAME", asnumpy=False, atleast=3
-    )
-    logger.info(fence.dataframe.to_string())
-
-    assert len(fence.dataframe) == 9
-    assert fence.dataframe.H_DELTALEN.mean() == pytest.approx(3.6, abs=0.02)
-    assert fence.dataframe.H_DELTALEN.std() <= 0.001
-
-
-def test_fence_from_more_slanted_polygon():
-    """Test fence from poly which is slanted; and total HLEN is > than distance"""
-
-    pol = Polygons()
-
-    mypoly = {
-        pol.xname: [0.0, 7.0, 15.0],
-        pol.yname: [100.0, 110.0, 120.0],
-        pol.zname: [0, 50, 60],
-        pol.pname: [1, 1, 1],
-    }
-    pol.dataframe = pd.DataFrame(mypoly)
-
-    fence = pol.get_fence(
-        distance=10, nextend=1, name="SOMENAME", asnumpy=False, atleast=3
-    )
-    logger.info(fence.dataframe.to_string())
-
-    assert len(fence.dataframe) == 5
-    assert fence.dataframe.H_DELTALEN.mean() == pytest.approx(12.49, abs=0.02)
-    assert fence.dataframe.H_DELTALEN.std() <= 0.001
-
-
-def test_rename_columns():
-    """Renaming xname, yname, zname"""
-
-    pol = Polygons(POLSET2)
-    assert pol.xname == "X_UTME"
+    poi = Points([(4.0, 4.0, 0.0), (4.5, 4.0, 0.0), (7.0, 7.0, 0.0), (20.0, 5.0, 0.0)])
+    poi.operation_polygons(pol, value=1, opname="eli", inside=False)
+    assert len(poi.dataframe) == 0
 
-    pol.xname = "NEWX"
-    assert pol.xname == "NEWX"
+def test_check_multi_point_multi_polyon_outside_op():
```

Comment:
This test puzzled me slightly, as I couldn't understand why running the `operation_polygons` would work on the exactly opposite number of points when swapping `inside`. But it turns out it's looping over the polygons, hence using `inside=False` with more than a single polygon simply doesn't make sense?

Reply:
It is stated in the docstring though:

>     'Inside' several polygons will become a union, while 'outside' polygons
>     will be the intersection.